scene/gui/option_button.cpp

```diff
@@ -435,9 +435,26 @@ void OptionButton::_select(int p_which, bool p_emit) {
 }
 
 void OptionButton::_select_int(int p_which) {
-	if (p_which < NONE_SELECTED || p_which >= popup->get_item_count()) {
+	if (p_which < NONE_SELECTED) {
 		return;
 	}
+
+	if (p_which >= popup->get_item_count()) {
+		if (initialized) {
+			ERR_FAIL_INDEX(current, popup->get_item_count());
+			return;
+		} else {
+			queued_current = p_which;
+		}
+	}
+
+	if (!initialized) {
+		if (queued_current != current) {
+			current = queued_current;
+		}
+		initialized = true;
+	}
```

Comment:
My bad I deleted that suggestion as I got confused about the order, sorry to confuse there didn't realize you saw it and responded

Reply:
It does still work so should I just keep it like that?

Reply:
Restore it to how it was, it was correct as far as I could tell